code/modules/client/preference_setup/general/03_flavor.dm

```diff
@@ -35,27 +43,27 @@
 		switch(href_list["flavour_text_robot"])
 			if("open")
 			if("Default")
-				var/msg = sanitize(input_cp1251(usr,"Set the default flavour text for your robot. It will be used for any module without individual setting.","Flavour Text",pref.flavour_texts_robot["Default"], "message"), extra = 0)
+				var/msg = sanitize(input(usr,"Set the default flavour text for your robot. It will be used for any module without individual setting.","Flavour Text",html_decode(pref.flavour_texts_robot["Default"])) as message, extra = 0)
 				if(CanUseTopic(user))
 					pref.flavour_texts_robot[href_list["flavour_text_robot"]] = msg
 			else
-				var/msg = sanitize(input_cp1251(usr,"Set the flavour text for your robot with [href_list["flavour_text_robot"]] module. If you leave this empty, default flavour text will be used for this module.","Flavour Text", pref.flavour_texts_robot[href_list["flavour_text_robot"]], "message"), extra = 0)
+				var/msg = sanitize(input(usr,"Set the flavour text for your robot with [href_list["flavour_text_robot"]] module. If you leave this empty, default flavour text will be used for this module.","Flavour Text",html_decode(pref.flavour_texts_robot[href_list["flavour_text_robot"]])) as message, extra = 0)
```

Comment:
cp encoding was removed on purpose?

Reply:
keep in mind that most of the code is copypasted and i cant notice all the changes, will fix